backends/backend_openssl3.c

```diff
@@ -2472,3 +2472,629 @@ static void openssl_ecdsa_backend(void)
 {
 	register_ecdsa_impl(&openssl_ecdsa);
 }
+
+static int rsa_create_pkey(EVP_PKEY **pkey,
+						const unsigned char *n, size_t n_len,
+						const unsigned char *e, size_t e_len,
+						const unsigned char *d, size_t d_len,
+						BN_CTX *bn_ctx)
+{
+	int ret = 0;
+	EVP_PKEY_CTX *ctx = NULL;
+	OSSL_PARAM_BLD *bld = NULL;
+	OSSL_PARAM *params = NULL;
+	unsigned char *n1 = NULL, *e1 = NULL, *d1 = NULL;
+	size_t n_len1 = 0, e_len1 = 0, d_len1 = 0;
+	BIGNUM *e_bn = NULL, *n_bn = NULL, *d_bn= NULL;
+	(void) bn_ctx;
+	bld = OSSL_PARAM_BLD_new();
+	n_bn = BN_CTX_get(bn_ctx);
+	BN_bin2bn(n, n_len, n_bn);
+	OSSL_PARAM_BLD_push_BN(bld, OSSL_PKEY_PARAM_RSA_N, n_bn);
+
+	if (e != NULL) {
+		e_bn = BN_CTX_get(bn_ctx);
+		BN_bin2bn(e, e_len, e_bn);
+		OSSL_PARAM_BLD_push_BN(bld, OSSL_PKEY_PARAM_RSA_E, e_bn);
+	}
+	if (d != NULL) {
+		d_bn = BN_CTX_get(bn_ctx);
+		BN_bin2bn(d, d_len, d_bn);
+		OSSL_PARAM_BLD_push_BN(bld, OSSL_PKEY_PARAM_RSA_D, d_bn);
+	}
+
+	params = OSSL_PARAM_BLD_to_param(bld);
+	ctx = EVP_PKEY_CTX_new_from_name(NULL, "RSA", NULL);
```

Comment:
flag "fips=yes" is not set. Is this okay ?

Reply:
[Force to fips mode if available](https://github.com/sikkamukul/acvpparser/blob/bae9a0243215af39102a3f542411cc7ce48da23a/backends/backend_openssl3.c#L34)